bson/src/test/unit/org/bson/codecs/VectorCodecTest.java

```diff
@@ -0,0 +1,162 @@
+
+    private static Stream<Arguments> provideVectorsAndCodecsForRoundTrip() {
+        return Stream.of(
+                arguments(Vector.floatVector(new float[]{1.1f, 2.2f, 3.3f}), new Float32VectorCodec(), Float32Vector.class),
+                arguments(Vector.int8Vector(new byte[]{10, 20, 30, 40}), new Int8VectorCodec(), Int8Vector.class),
+                arguments(Vector.packedBitVector(new byte[]{(byte) 0b10101010, (byte) 0b01010101}, (byte) 3), new PackedBitVectorCodec(), PackedBitVector.class),
+                arguments(Vector.packedBitVector(new byte[]{(byte) 0b10101010, (byte) 0b01010101}, (byte) 3), new VectorCodec(), Vector.class),
+                arguments(Vector.int8Vector(new byte[]{10, 20, 30, 40}), new VectorCodec(), Vector.class),
+                arguments(Vector.packedBitVector(new byte[]{(byte) 0b10101010, (byte) 0b01010101}, (byte) 3), new VectorCodec(), Vector.class)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldRoundTripVectors(final Vector vectorToEncode) {
+        //given
+        Document expectedDocument = new Document("vector", vectorToEncode);
+
+        //when
+        Codec<Document> codec = CODEC_REGISTRIES.get(Document.class);
+        OutputBuffer buffer = encode(codec, expectedDocument);
+        Document actualDecodedDocument = decode(codec, buffer);
+
+        //then
+        Binary binaryVector = (Binary) actualDecodedDocument.get("vector");
+        assertNotEquals(actualDecodedDocument, expectedDocument);
+        Vector actualVector = binaryVector.asVector();
+        assertEquals(actualVector, vectorToEncode);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldEncodeVector(final Vector vectorToEncode, final Codec<Vector> vectorCodec) {
+        // given
+        BsonWriter mockWriter = Mockito.mock(BsonWriter.class);
+
+        // when
+        vectorCodec.encode(mockWriter, vectorToEncode, EncoderContext.builder().build());
+
+        // then
+        verify(mockWriter, times(1)).writeBinaryData(new BsonBinary(vectorToEncode));
+        verifyNoMoreInteractions(mockWriter);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideVectorsAndCodecsForRoundTrip")
+    void shouldDecodeVector(final Vector vectorToDecode, final Codec<Vector> vectorCodec) {
+        // given
+        BsonReader mockReader = Mockito.mock(BsonReader.class);
+        BsonBinary bsonBinary = new BsonBinary(vectorToDecode);
+        when(mockReader.peekBinarySubType()).thenReturn(BsonBinarySubType.VECTOR.getValue());
+        when(mockReader.readBinaryData()).thenReturn(bsonBinary);
+
+        // when
+        Vector decodedVector = vectorCodec.decode(mockReader, DecoderContext.builder().build());
+
+        // then
+        assertNotNull(decodedVector);
+        assertEquals(vectorToDecode, decodedVector);
+    }
```

Comment:
- Given that we have `shouldRoundTripVectors`, what value do these two tests add, and why do they need to use `Mockito`?
- If it turns out that these tests are actually useful and mocking is necessary, we should use `MongoMockito`.

Reply:
Good point. Upon reviewing it, the round-trip test should actually be in `DocumentCodecTest` rather than `VectorCodecTest`, as it’s specifically testing that encoding and decoding a `Document` uses a `Vector` codecs by default when `Vector` object is present. I have already extended the existing round-trip test in `DocumentCodecTest` for that case.

Regarding the tests in `VectorCodecTest`, I’ve kept encode and decode but replaced the behavior verification approach (mocking) with a state verification approach that doesn’t rely on mocks.